In [31]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [32]:
url = 'https://github.com/nytimes/covid-19-data/blob/master/us-counties-2020.csv?raw=true'
df = pd.read_csv(url)

In [33]:
df['date'] = pd.to_datetime(df['date'])

In [34]:
df_first_wave = df[df['date'] < '06/30/2020']

In [35]:
treatment_states = ['Minnesota', 'Montana','Nevada','North Carolina','Rhode Island', 'Alabama','Arizona','Florida','Georgia','Kansas','Maine','Maryland','Mississippi','Missouri','New Hampshire','South Carolina','Tennesse','Texas','Utah','Virgina']

In [36]:
control_states = ['Arkansas','Iowa','Nebraska','North Dakota','Oklahoma', 'South Dakota', 'Wyoming']

In [37]:
treatment_df = df_first_wave[df_first_wave['state'].isin(treatment_states)]

In [38]:
control_df = df_first_wave[df_first_wave['state'].isin(control_states)]

In [39]:
control_df['Treatment'] = False

/tmp/ipykernel_5851/1425731934.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  control_df['Treatment'] = False


In [40]:
treatment_df['Treatment'] = True

/tmp/ipykernel_5851/283517631.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  treatment_df['Treatment'] = True


In [41]:
treatment_df

date      county    state     fips  cases  deaths  Treatment
8      2020-01-26    Maricopa  Arizona   4013.0      1     0.0       True
13     2020-01-27    Maricopa  Arizona   4013.0      1     0.0       True
18     2020-01-28    Maricopa  Arizona   4013.0      1     0.0       True
23     2020-01-29    Maricopa  Arizona   4013.0      1     0.0       True
28     2020-01-30    Maricopa  Arizona   4013.0      1     0.0       True
...           ...         ...      ...      ...    ...     ...        ...
286736 2020-06-29     Unknown     Utah      NaN      0     9.0       True
286737 2020-06-29        Utah     Utah  49049.0   3965    22.0       True
286738 2020-06-29     Wasatch     Utah  49051.0    406     3.0       True
286739 2020-06-29  Washington     Utah  49053.0   1156     0.0       True
286740 2020-06-29       Weber     Utah  49057.0    910    12.0       True

[111192 rows x 7 columns]

In [42]:
combined_df = pd.concat([control_df, treatment_df])
combined_df

date      county     state     fips  cases  deaths  Treatment
198    2020-02-17     Douglas  Nebraska  31055.0     10     0.0      False
210    2020-02-18     Douglas  Nebraska  31055.0     10     0.0      False
222    2020-02-19     Douglas  Nebraska  31055.0     10     0.0      False
235    2020-02-20     Douglas  Nebraska  31055.0     11     0.0      False
249    2020-02-21     Douglas  Nebraska  31055.0     11     0.0      False
...           ...         ...       ...      ...    ...     ...        ...
286736 2020-06-29     Unknown      Utah      NaN      0     9.0       True
286737 2020-06-29        Utah      Utah  49049.0   3965    22.0       True
286738 2020-06-29     Wasatch      Utah  49051.0    406     3.0       True
286739 2020-06-29  Washington      Utah  49053.0   1156     0.0       True
286740 2020-06-29       Weber      Utah  49057.0    910    12.0       True

[150050 rows x 7 columns]

In [44]:
combined_df['county'].value_counts()

county
Washington    1607
Lincoln       1244
Clay          1227
Jackson       1202
Unknown       1157
              ... 
Granite          5
Hettinger        4
Elk              4
Eureka           4
Rush             1
Name: count, Length: 1121, dtype: int64

## Dropping values where 'county' is 'Unknown' as there is no way to impute. Losing 1157 records of 150050 initial total (0.77%)

In [48]:
#drop unknown county
combined_df = combined_df[combined_df['county'] != 'Unknown']
combined_df

date      county     state     fips  cases  deaths  Treatment
198    2020-02-17     Douglas  Nebraska  31055.0     10     0.0      False
210    2020-02-18     Douglas  Nebraska  31055.0     10     0.0      False
222    2020-02-19     Douglas  Nebraska  31055.0     10     0.0      False
235    2020-02-20     Douglas  Nebraska  31055.0     11     0.0      False
249    2020-02-21     Douglas  Nebraska  31055.0     11     0.0      False
...           ...         ...       ...      ...    ...     ...        ...
286735 2020-06-29      Uintah      Utah  49047.0     30     0.0       True
286737 2020-06-29        Utah      Utah  49049.0   3965    22.0       True
286738 2020-06-29     Wasatch      Utah  49051.0    406     3.0       True
286739 2020-06-29  Washington      Utah  49053.0   1156     0.0       True
286740 2020-06-29       Weber      Utah  49057.0    910    12.0       True

[148893 rows x 7 columns]